src/sgl/func/base_struct.cpp

```
@@ -0,0 +1,50 @@
+{
+    SGL_CHECK(m_type, "BaseStruct has no semantic type");
+    return m_type->full_name();
+}
+
+const slangpy::Shape& BaseStruct::shape() const
+{
+    SGL_CHECK(m_type, "BaseStruct has no semantic type");
+    return m_type->shape();
+}
+
+void BaseStruct::on_hot_reload(ref<const TypeReflection> type_reflection)
```

Cppcheck (2.20.0)

[style] The function 'on_hot_reload' is never used.

(unusedFunction)